File: src/main/java/link/rdcn/user/UsernamePassword.java
package link.rdcn.user;

public class UsernamePassword extends Credentials {
    private final String username;
    private final String password;

    public UsernamePassword(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }
}

File: src/main/scala/link/rdcn/user/UsernamePassword.scala
package link.rdcn.user


/**
 * @Author renhao
 * @Description:
 * @Data 2025/7/1 15:04
 * @Modified By:
 */
case class UsernamePassword(userName: String, password: String) extends Credentials {
  override def isAnonymous: Boolean = this == Credentials.ANONYMOUS
}